### ISSUE TITLE
Running ParmEd with jupyter-notebook

### ISSUE THREAD

**Issue by slochower** [ParmEd/ParmEd#747]
Hi, I'm having difficulty running ParmEd within a jupyter-notebook and I'm unclear if what I'm seeing is the intended behavior. I'm following the instructions for using the API here http://parmed.github.io/ParmEd/html/parmed_api.html where the first code block shows an example of how to import `parmed.tools`.

If I start a python 2.7 console, I can import `parmed.amber` without trouble and enter ParmEd actions at the prompt.

```python
>>> import sys
>>> print(sys.version)
2.7.5 (default, Nov 20 2015, 02:00:19) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
>>> from parmed.amber import AmberParm

 _~__________________
 |_|_______________|_|
 |  ,-------------.  |
 | |  .---------.  | |
 | |  |P        |  | |
 | |  | a       |  | |
 | |  |  r      |  | |
 | |  |   m  ED |  | |
 | |  `---------'  | |
 | `---------------' |
 |   _ ParmedBoy     |
 | _| |_         ,-. |
 ||_ O _|   ,-. "._,"|
 |  |_|    "._,"   A |
 |    _  _    B      |
 |   // //           |
 |  // //    \\\\\\  |
 |  `  `      \\\\\\ ,
 |________...______,"
 
ParmEd: a Parameter file Editor


Reading input from STDIN...
> 
```

If I try to run `from parmed.amber import AmberParm` in jupyter-notebook, I get an error:
```python
In [1] import sys
       print(sys.version)
       sys.path.insert(0, '/home/xxx/amber14gnu/AmberTools/src/parmed')

2.7.12 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:42:40) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]

In [8]: from parmed.amber import AmberParm
usage: __main__.py [-h] [-v] [-i FILE] [-p <prmtop>] [-c <inpcrd>] [-O]
                   [-l FILE] [--prompt PROMPT] [-n] [-e] [-s] [-r]
                   [<prmtop>] [<script>]
__main__.py: error: unrecognized arguments: -f
An exception has occurred, use %tb to see the full traceback.

SystemExit: 2

In [11]: %tb
SystemExit                                Traceback (most recent call last)
<ipython-input-10-de61dfce161f> in <module>()
----> 1 from parmed.amber import AmberParm

/home/xxx/amber14gnu/AmberTools/src/parmed/parmed.py in <module>()
     94          help='File with a series of ParmEd commands to execute.')
     95 
---> 96 opt = parser.parse_args()
     97 
     98 # If the user specified a prmtop and script in the 'old' way, append them to the

/home/xxx/anaconda3/envs/ipykernel_py2/lib/python2.7/argparse.pyc in parse_args(self, args, namespace)
   1702         if argv:
   1703             msg = _('unrecognized arguments: %s')
-> 1704             self.error(msg % ' '.join(argv))
   1705         return args
   1706 

/home/xxx/anaconda3/envs/ipykernel_py2/lib/python2.7/argparse.pyc in error(self, message)
   2372         """
   2373         self.print_usage(_sys.stderr)
-> 2374         self.exit(2, _('%s: error: %s\n') % (self.prog, message))

/home/xxx/anaconda3/envs/ipykernel_py2/lib/python2.7/argparse.pyc in exit(self, status, message)
   2360         if message:
   2361             self._print_message(message, _sys.stderr)
-> 2362         _sys.exit(status)
   2363 
   2364     def error(self, message):

SystemExit: 2
```

I suspect this is somehow related to the fact that jupyter-notebook (as far as I know) won't give me the ParmEd prompt. Is there a more appropriate way to import the ParmEd modules so I can create an `action` and then do `action.execute()`?

**Comment by hainm** [comment]
which ParmEd version you are using? I suggest to use ParmEd version in AmberTools 16 or you can install from [github](https://github.com/ParmEd/ParmEd#installing-parmed)

But you are right that using Ipython or Jupyter notebook won't give you the ParmEd prompt.

You can access ParmEd's action via `parmed.tools`

![screenshot-3](https://cloud.githubusercontent.com/assets/4451957/18015497/fd0f0cf8-6b95-11e6-9268-b9db9b078ee6.png)

**Comment by hainm** [comment]
```python
In [1]: import parmed as pmd

In [2]: parm = pmd.load_file('./ash.parm7')

In [3]: pmd.tools.summary(parm)
Out[3]: 
Amino Acid Residues:   1
Nucleic Acid Residues: 0
Number of cations:     0
Number of anions:      0
Num. of solvent mols:  0
Num. of unknown res:   2
Total charge (e-):     0.0000
Total mass (amu):      188.1860
Number of atoms:       25
Number of residues:    3
Residue set:           ACE, ASH, NME
Residue count:         ACE: 1, ASH: 1, NME: 1
```

and so on.

**Comment by slochower** [comment]
Ah, right. After I install ParmEd from github, I can `import parmed` in jupyter-notebook just like your last comment. Earlier I was using version 15.0b. Thanks!

**Comment by swails** [comment]
Just a quick explanation as to what you saw:

In its earliest days, `parmed.py` was the front-end program wrapping the classes I wrote in the `chemistry` package.

As adoption increased and it started to play nicely with other Python packages in the comp chem ecosystem, I changed the package name to `parmed/` and had to remove the `.py` suffix on `parmed.py` to prevent it from being importable.

So when you typed `import parmed` in Jupyter with the old version of ParmEd, it was importing the top-level script and (since none of it was protected inside an `if __name__ == '__main__':`) executing it.  Not something you want to do inside an interactive interpreter :).
